### PR TITLE
tctl 1.5.1 (new formula)

### PR DESCRIPTION
**Pull request by Sushisource**
Adds a formula for the Temporal (https://temporal.io/) command line tool.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/tctl.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  desc "Temporal CLI (tctl)"
+  homepage "https://temporal.io/"
+  url "https://github.com/temporalio/temporal/archive/v1.5.1.tar.gz"
+  sha256 "a2f75873d7c211e07c40653c605e64fcf732543c9d50683b2d61a985f6b3f37f"
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
```
Comment (carlocab) [inline Formula/tctl.rb:11]:
```suggestion
    ENV["GOPATH"] = buildpath
    ENV["GO111MODULE"] = "auto"
```
I think you need this to compile on ARM with Go 1.16. However, let's wait and see if it's needed first.

File: Formula/tctl.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    bin_path = buildpath/"src/temporal.io/tctl_staging"
+    # Copy all files from their current location (GOPATH root) to the binpath
+    bin_path.install Dir["*"]
+    cd bin_path do
+      # Install the compiled binary into Homebrew's `bin`
+      system "go", "build", "-o", bin/"tctl", "cmd/tools/cli/main.go"
```
Comment (carlocab) [inline Formula/tctl.rb:18]:
```suggestion
      system "go", "build", *std_go_args, "cmd/tools/cli/main.go"
```

File: Formula/tctl.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    end
+  end
+
+  test do
+    # Given tctl is pointless without a server, not much intersting to test here.
+    run_output = shell_output("#{bin}/tctl --version 2>&1")
+    assert_match "tctl version", run_output
```
Comment (carlocab) [inline Formula/tctl.rb:25]: Can you try running a command that you know will fail, then check that the failure message is what you expect?

**Review by carlocab — COMMENTED** [review]
Thanks for your PR, @Sushisource. A few comments.

File: Formula/tctl.rb (hunk 1 above)
Comment (carlocab) [reply Formula/tctl.rb:11]: Ok, doesn't look like this one is needed actually.

**Comment by Sushisource** [comment]
@carlocab Fixed, thanks!

**Review by carlocab — DISMISSED** [review]

**Review by chenrui333 — DISMISSED** [review]

File: Formula/tctl.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/tctl.rb:11]: Why do you need GOPATH as you already have the go mod builds?

File: Formula/tctl.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  def install
+    ENV["GOPATH"] = buildpath
+
+    bin_path = buildpath/"src/temporal.io/tctl_staging"
+    # Copy all files from their current location (GOPATH root) to the binpath
+    bin_path.install Dir["*"]
+    cd bin_path do
+      # Install the compiled binary into Homebrew's `bin`
+      system "go", "build", *std_go_args, "cmd/tools/cli/main.go"
+    end
```
Comment (chenrui333) [inline Formula/tctl.rb:19]: This can be updated to use go modules

Comment (bayandin) [inline Formula/tctl.rb:19]:
```suggestion
    system "go", "build", *std_go_args, "-ldflags", "-s -w", "./cmd/tools/cli/main.go"
```

Comment (Sushisource) [reply Formula/tctl.rb]: Thanks!

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@Sushisource, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request closed without merging**